cpp/src/strings/search/findall.cu

```diff
@@ -21,134 +21,148 @@
-  std::unique_ptr<column> operator()(column_device_view const& d_strings,
-                                     size_type column_index,
-                                     size_type const* d_find_counts,
-                                     rmm::cuda_stream_view stream,
-                                     rmm::mr::device_memory_resource* mr)
+  void operator()(column_device_view const& d_strings,
+                  size_type const* d_find_counts,
+                  indices_span& d_indices,
+                  rmm::cuda_stream_view stream)
   {
-    rmm::device_uvector<string_index_pair> indices(d_strings.size(), stream);
-
-    thrust::transform(rmm::exec_policy(stream),
-                      thrust::make_counting_iterator<size_type>(0),
-                      thrust::make_counting_iterator<size_type>(d_strings.size()),
-                      indices.begin(),
-                      findall_fn<stack_size>{d_strings, d_prog, column_index, d_find_counts});
-
-    return make_strings_column(indices.begin(), indices.end(), stream, mr);
+    thrust::for_each_n(rmm::exec_policy(stream),
+                       thrust::make_counting_iterator<size_type>(0),
+                       d_strings.size(),
+                       findall_fn<stack_size>{d_strings, d_prog, d_find_counts, d_indices});
   }
 };
 }  // namespace
 
 //
-std::unique_ptr<table> findall(
-  strings_column_view const& input,
-  std::string const& pattern,
-  regex_flags const flags,
-  rmm::cuda_stream_view stream,
-  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource())
+std::unique_ptr<table> findall(strings_column_view const& input,
+                               std::string const& pattern,
+                               regex_flags const flags,
+                               rmm::cuda_stream_view stream,
+                               rmm::mr::device_memory_resource* mr)
 {
   auto const strings_count = input.size();
-  auto const d_strings     = column_device_view::create(input.parent(), stream);
 
   // compile regex into device object
   auto const d_prog =
     reprog_device::create(pattern, flags, get_character_flags_table(), strings_count, stream);
-  auto const regex_insts = d_prog->insts_counts();
 
-  auto find_counts =
-    count_matches(*d_strings, *d_prog, stream, rmm::mr::get_current_device_resource());
-  auto d_find_counts = find_counts->mutable_view().data<size_type>();
-
-  std::vector<std::unique_ptr<column>> results;
+  auto const d_strings = column_device_view::create(input.parent(), stream);
+  auto find_counts     = count_matches(*d_strings, *d_prog, stream);
+  auto d_find_counts   = find_counts->view().data<size_type>();
 
-  size_type const columns = thrust::reduce(
+  size_type const columns_count = thrust::reduce(
     rmm::exec_policy(stream), d_find_counts, d_find_counts + strings_count, 0, thrust::maximum{});
 
+  auto indices = rmm::device_uvector<string_index_pair>(strings_count * columns_count, stream);
+
```

Comment:
```suggestion
```

Reply:
This is also an intentional separation between lines 129 and 131. I prefer this for separation because even though this is a single line of code it is a large memory allocation and very significant step in the logic of the function. I don't see how removing the blank line makes the code any clearer.